lib/auth/in_memory/checker.js

```diff
@@ -0,0 +1,195 @@
+    }
+    if (checkSas &&
+        checkExists(data, 'Account', 'sasToken', 'string', accountObj)) {
+        // eslint-disable-next-line no-param-reassign
+        data.sasTokens[accountObj.sasToken] =
+            incr(data.sasTokens[accountObj.sasToken]);
+    }
+
+    if (accountObj.users) {
+        if (checkType(data, 'Account', 'users', 'array', accountObj)) {
+            accountObj.users.forEach(userObj => checkUser(data, userObj));
+        }
+    }
+
+    if (accountObj.keys) {
+        if (checkType(data, 'Account', 'keys', 'array', accountObj)) {
+            accountObj.keys.forEach(keyObj => {
+                // eslint-disable-next-line no-param-reassign
+                data.keys[keyObj.access] = incr(data.keys[keyObj.access]);
+            });
+        }
+    }
+}
+
+function dumpCountError(property, obj, log) {
+    let count = 0;
+    Object.keys(obj).forEach(key => {
+        if (obj[key] > 1) {
+            log.error('property should be unique', {
+                property,
+                value: key,
+                count: obj[key],
+            });
+            ++count;
+        }
+    });
+    return count;
+}
+
+function dumpErrors(checkData, log) {
+    let nerr = dumpCountError('CanonicalID', checkData.canonicalIds, log);
+    nerr += dumpCountError('Email', checkData.emails, log);
+    nerr += dumpCountError('ARN', checkData.arns, log);
+    nerr += dumpCountError('AccessKey', checkData.keys, log);
+    nerr += dumpCountError('SAS Token', checkData.sasTokens, log);
+
+    if (checkData.errors.length > 0) {
+        checkData.errors.forEach(msg => {
+            log.error(msg.txt, msg.obj);
+        });
+    }
+
+    if (checkData.errors.length === 0 && nerr === 0) {
+        return false;
+    }
+
+    log.fatal('invalid authentication config file (cannot start)');
+
+    return true;
+}
+
+/**
+ * @param {object} authdata - the authentication config file's data
+ * @param {object} loggerConfig - config for logger if needed to dump errors
+ * @param {string} loggerConfig.level - logger level
+ * @param {string} loggerConfig.dump - logger dump level
+ * @param {(boolean|null)} checkSas - whether to check Azure SAS for ea. account
+ * @return {boolean} true on erroneous data
+ *                   false on success
+ */
+function check(authdata, loggerConfig, checkSas) {
+    const checkData = {
+        errors: [],
+        emails: [],
+        arns: [],
+        canonicalIds: [],
+        keys: [],
+        sasTokens: [],
+    };
+    const log = new werelogs.Logger('S3', loggerConfig);
+
+
+    if (authdata.accounts === undefined) {
+        checkData.errors.push({
+            txt: 'no "accounts" array defined in Auth config',
+        });
+        return dumpErrors(checkData, log);
+    }
+
+    authdata.accounts.forEach(account => {
+        checkAccount(checkData, account, checkSas);
+    });
+
+    return dumpErrors(checkData, log);
+}
+
+module.exports = check;
```

Comment:
I think `check` is too generic, may be call this validateAuthConfig or something like that. Make sure your filename reflects the same name as the method you are exporting.